### PR TITLE
chore: Some minor changes to the docs linter

### PR DESCRIPTION
**Pull request by jannfis**
Some minor changes to the docs linter:

* Moves `/white-list` to `/docs/url-allow-list`
* Includes a new URL to allow linking Argo CD Helm chart
 
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by jannfis** [comment]
On a minor note, we probably should mention somewhere in the docs that the docs are linted, and if people introduce new URLs in the official docs, they will need to include that URL in the `docs/url-allow-list` as well to satisfy the docs linter.

**Comment by jannfis** [comment]
This is still failing, looking into it

File: docs/url-allow-list [hunk 1]
```diff
@@ -66,3 +66,4 @@ ghe.example.com
 192.168.99.100:8443
 github.com/yourghuser/argo-cd
 github.com/argoproj/argo-cd/releases/download/
+github.com/argoproj/argo-helm/tree/master/charts/argo-cd
```
Comment (crenshaw-dev) [inline docs/url-allow-list:69]: Why is it necessary to allowlist this? It resolves fine for me, the the URL in installation.md matches this exactly.

Comment (jannfis) [reply docs/url-allow-list:69]: My bad. It failed on another URL in the file, which I hadn't noticed before.

**Comment by crenshaw-dev** [comment]
That failure is weird. Not sure why the linter seems to be picking up a semicolon on the end of the URL.

**Comment by jannfis** [comment]
I simply chose to include the URL with `;`-suffix in the whitelist now.

**Review by alexmt — APPROVED** [review]
Thank you for fixing it!

**Pull request merged**